src/sys/mman.rs

```diff
@@ -187,6 +190,26 @@ libc_bitflags! {
     }
 }
 
+impl MapFlags {
+    /// Create `MAP_HUGETLB` with provided size of huge page.
+    ///
+    /// Under the hood it computes `MAP_HUGETLB | (huge_page_size_log2 as i32) << libc::MAP_HUGE_SHIFT`.
+    /// `huge_page_size_log2` denotes logarithm of huge page size to use and should between 16 and 63
+    /// (inclusively). For example, `MapFlags::map_hugetlb_with_shift(30)` returns value equal to
+    /// `Some(MapFlags::MAP_HUGETLB | MAP_HUGE_1GB)`.
+    #[cfg(target_os = "linux")]
+    #[cfg_attr(docsrs, doc(cfg(all())))]
+    pub fn map_hugetlb_with_shift(huge_page_size_log2: u8) -> Option<Self> {
```

Comment:
I couldn't make it `const` because I couldn't find a way to convert `i32` into `InternalBitFlags`.